### PR TITLE
disable tf GPU visibility

### PR DESCRIPTION
**Pull request by keshavb96**
Maxtext crashes when running with TFDS data with a GPU OOM. This is traced back to Tensorflow spawning processes that allocate memory on GPUs. This patch is a current work around to hide GPUs from TF.

### CODE REVIEW

**Review by wang2yn84 — APPROVED** [review]
Thank you! Just to keep the record, it aligns with https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth.

File: MaxText/train.py [hunk 1]
```diff
@@ -755,6 +756,9 @@ def train_loop(config, state=None):
 
 def main(argv: Sequence[str]) -> None:
   jax.config.update("jax_default_prng_impl", "unsafe_rbg")
+  # TF allocates extraneous GPU memory when using TFDS data
+  # this leads to CUDA OOMs. WAR for now is to hide GPUs from TF
```
Comment (gobbleturk) [inline MaxText/train.py:760]: What does WAR here mean? I assume its an acronym?

**Review by gobbleturk — APPROVED** [review]

File: MaxText/train.py (hunk 1 above)
Comment (wang2yn84) [reply MaxText/train.py:760]: Work around I guess?

Comment (nouiz) [reply MaxText/train.py:760]: Exact.

**Pull request merged**